File: laravel/laravel/code/app/Console/Commands/filechecksum/enconfig.py
#!/usr/bin/python
__author__ = "Vickyraj Chavan"
__date__ = "03-06-2019"
__last_updated__= "25 FEB 2020"
#========================== IMPORT ETC ==========================
import db
import  sys, MySQLdb, os, requests, json, ast
from lite_logging import log

# db connection
dbobj = db.mysql()

log  = log()
# get the dynamic log file name
file_name = os.path.basename(__file__)
file_name = os.path.splitext(file_name)[0]
logfile = file_name+".log"
tid =1
ensysconfig = 'ensysconfig'
global config_data
config_data = {}
dataarray = {}
dataarray['functionality'] = "enconfig"
#get details from system_settings and store it in dictionary
def get_config():
    try:
        dataarray['function_name'] = "get_config"
        config_ = dbobj.select_dict("select configuration from en_system_settings where type='ensysconfig' AND status='y'",None,True)    
        conf_data =  config_["configuration"]
        config_data = json.loads(conf_data)
        return config_data
    except Exception, e:
	log.u(tid, "[ Error in system configuration : %s]" %e ,logfile)
        sys.exit() 


def load_config(config_data,config_item = ""):    
    try:
    	dataarray['function_name'] = "load_config"
        config_item = "all"
        from_ = config_data['en_sysconfig_config']
        from_ = from_ if from_ != '' else 'api'
        if from_ == "api" :           
            url = config_data['en_sysconfig_api_url']+'/config/get?config_item='+config_item
            config_data_ = requests.get(url = url)
            config_data_ = config_data_.json()
            if config_data_ :
                config_data_ = process_config(config_data_)
            else:
		log.u(tid,"No response from API",logfile)
                sys.exit("No response from API")
        elif from_ == "local":                    
            if not os.path.isdir(config_data['en_sysconfig_config_path']):
                try:
                    os.makedirs(config_data['en_sysconfig_config_path'])
                except OSError as e:
                    if e.errno != errno.EEXIST:
                    	log.u(tid,"Unable to create system directory",logfile)
                        raise("Unable to create system directory")
            if os.path.exists(config_data['en_sysconfig_config_path']+"/"+ensysconfig):    
                f = open(config_data['en_sysconfig_config_path']+"/"+ensysconfig, "r")
                data = f.readline()
                if data != "" :
                    data = json.dumps(data)
                    data = ast.literal_eval(data)
                    return data         
            else :       
                url = config_data['en_sysconfig_api_url']+'/config/get?config_item='+config_item
                config_data_ = requests.get(url = url)
                if config_data_ :
                    config_data_ = config_data_.json()
                    config_data_ = process_config(config_data_)
                    config_data_ = json.dumps(config_data_)
                    try:                
                        f = open(config_data['en_sysconfig_config_path']+"/"+ensysconfig, "a")
                        f.write(str(config_data_))
                        f.close()
                    except IOError as x:
                        if x.errno == errno.ENOENT:
				log.u(tid,"Directory or file does not exist : %s " %x,logfile)                              
				sys.exit("Directory or file does not exist => "+config_data['en_sysconfig_config_path']+"/"+ensysconfig)
				
                        elif x.errno == errno.EACCES:   
				log.u(tid,"Directory or file is not writable :%s " %x,logfile)         
                            	sys.exit("Directory or file is not writable => "+config_data['en_sysconfig_config_path']+"/"+ensysconfig)
                        else:
                        	sys.exit(argv[1], '- some other error')                
                else:
			log.u(tid,"No response from API",logfile)
		        sys.exit("No response from API")
        config_data_ = json.dumps(config_data_)
        return config_data_
    except Exception, e:
		log.u(tid,"Error in loading message: %s " %e,logfile)
       		sys.exit()          

def process_config(response):
    try:
        dataarray['function_name'] = "process_config"
        if isinstance(response, dict) and response['status'] == "error":        
            sys.exit(response['message']['error'])        
        else:
            if response['status'] == "success" :            
                content = response['data']
                return content

    except Exception, e:
    	log.u(tid,"Error in processing configuration: %s " %e,logfile)
    	sys.exit() 

config_data = get_config()
#get_config()

File: laravel/laravel/code/app/Services/eNsysconfig/Enconfig.php
<?php
/*
Author: Vishal Chaudhari
Description: This library load the configuration from database and make configuration available to whole codeigniter
 */
/**
 * Use:
 *     $this->load->library('enconfig');
 *
 *
 **/
namespace App\Services\eNsysconfig;
use  App\Services\Restapi;
use  App\Services\RemoteApi;

class Enconfig
{
    /**
     * @param $ci
     */
    public function __construct($ci = true)
    {
        $this->item = '';
        $this->ensysconfig = 'ensysconfig';
        $this->item = $this->load_config("all", $ci); 
    }

    /**
     * @param $from
     * @param $config_item
     * @return mixed
     */
    public function load_config($config_item = "", $codeigniter = true)
    {
        global $system_settings, $restapiobj;
        try {
            if ($codeigniter)
            {
                //$this->ci = &get_instance();
                //$this->ci->load->library('restapi');
                $restapi = new Restapi;
                $from = config('app.en_sysconfig_config');
            }
            else
            {
                $from = $system_settings['en_sysconfig_config'];
            }

            $from = $from != '' ? $from : 'api';
            $config_ = $codeigniter ? config('enconfig') : $system_settings;
            $restapi = $codeigniter ? $restapi : $restapiobj;
            if ($from == "api")
            {
                apilog("+++++API++++++++");
               //                $url = $config_['en_sysconfig_api_url'].'/config/get?config_item='.$config_item;
                $url = config('app.en_sysconfig_api_url').'/config/get?config_item='.$config_item;
                $this->item = $restapi->apicall('get', $url);
                $this->item = $this->process_config($this->item);
            }
            else if ($from == "local")
            {
                apilog("+++++LOCAL++++++++");
                if (!is_dir(config('app.en_sysconfig_config_path')))
                {
                    if (!@mkdir(config('app.en_sysconfig_config_path'), 0755, true))
                    {
                        die("Unable to create system directory");
                    }
                }
                if (file_exists(config('app.en_sysconfig_config_path')."/".$this->ensysconfig))
                {
                    $data = file_get_contents(config('app.en_sysconfig_config_path')."/".$this->ensysconfig);
                    if (json_decode($data, true))
                    {
                        $this->item = json_decode($data, true);
                        return $this->item;
                    }
                }
                else
                {
                    // write file
                    $url = config('app.en_sysconfig_api_url').'/config/get?config_item='.$config_item;
                    $this->item = $restapi->apicall('get', $url);
                    $this->item = $this->process_config($this->item);
                    $config_key_value = false; // set false for json , set true for key value
                    $config_str = '';
                    if ($config_key_value)
                    {
                        if (is_array($this->item))
                        {
                            foreach ($this->item as $key => $val)
                            {
                                $config_str .= $key."=".$val."\n";
                            }
                        }
                        if ($config_str == '')
                        {
                            die(trans('messages.159'));
                        }
                    }
                    else
                    {
                        $config_str = json_encode($this->item);
                    }
                    if (!is_writable(config('app.en_sysconfig_config_path')))
                    {
                        die(trans('messages.155')." => ".config('app.en_sysconfig_config_path')."/".$this->ensysconfig);
                    }
                    $write = file_put_contents(config('app.en_sysconfig_config_path')."/".$this->ensysconfig, $config_str);
                    if (!$write)
                    {
                        die(trans('messages.157'));
                    }
                }
            }
        }
        catch (Exception $e)
        {
            $data['data'] = '';
            $data['message']['error'] = $e->message();
            $data['status'] = 'error';
           // echo response()->json($data);
            die();
        }
        catch (Error $e)
        {
            $data['data'] = '';
            $data['message']['error'] = $e->message();
            $data['status'] = 'error';
           // echo response()->json($data);
            die();
        }
        if ($codeigniter)
        {
            //$this->ci->config->set_item('language', _isset($this->item, 'language'));
           config('app.language', _isset($this->item, 'language'));

        }
        return $this->item;
    }
    /**
     * @param $response
     * @return mixed
     */
    public function process_config($response)
    {
        if (is_array($response) && isset($response['is_error']) && $response['is_error'])
        {
            die($response['msg']);
        }
        else
        {
            $content = _isset($response, 'content');
        }
        return $content;
    }
}